types/cheerio/index.d.ts

```diff
@@ -220,7 +220,8 @@ interface Cheerio {
 
     // Not Documented
 
-    toArray(): CheerioElement[];
+    // tslint:disable-next-line no-unnecessary-generics
+    toArray<T = CheerioElement>(): T[];
```

Comment:
This is considered an anti pattern.
Use a type assertion instead.

Reply:
Humm... Yeah, I do agree but the situation is the following:

I would like to have `MyNewElement` from `.toArray()`
```typescript
interface MyNewElement extends CheerioElement {
  myProperty: String
}
```

So I would do
```typescript
const myElements = $(...).map(... => ({
  myProperty: $(this).attr('class') // Just an example
}).toArray() as MyNewElement;
```

The problem is that `myElements` do not have the properties from `CheerioElement` anymore. They will have only `myProperty` inside. I need to extend `CheerioElement` to be able to do the casting but it doesn't actually extends the properties from `CheerioElement` once mapped -> array.

Am I missing something here?

Reply:
@SimonSchick ?